File: Language/C#/Book/tmp/LINQ实战/Chapter13.VB - LinqBooks/LinqBooks.Web/GetXML.ashx.vb
Imports System.Web
Imports System.Web.Services

Public Class GetXML
  Implements System.Web.IHttpHandler

  Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
    Dim dataSet = New LinqBooksDataSet()
    Dim subjectTableAdapter = New LinqBooksDataSetTableAdapters.SubjectTableAdapter()
    subjectTableAdapter.Fill(dataSet.Subject)
    Dim publisherTableAdapter = New LinqBooksDataSetTableAdapters.PublisherTableAdapter()
    publisherTableAdapter.Fill(dataSet.Publisher)
    Dim bookTableAdapter = New LinqBooksDataSetTableAdapters.BookTableAdapter()
    bookTableAdapter.Fill(dataSet.Book)
    Dim authorTableAdapter = New LinqBooksDataSetTableAdapters.AuthorTableAdapter()
    authorTableAdapter.Fill(dataSet.Author)
    Dim bookAuthorTableAdapter = New LinqBooksDataSetTableAdapters.BookAuthorTableAdapter()
    bookAuthorTableAdapter.Fill(dataSet.BookAuthor)
    Dim userTableAdapter = New LinqBooksDataSetTableAdapters.UserTableAdapter()
    userTableAdapter.Fill(dataSet.User)
    Dim reviewTableAdapter = New LinqBooksDataSetTableAdapters.ReviewTableAdapter()
    reviewTableAdapter.Fill(dataSet.Review)

    context.Response.AddHeader("Content-Disposition", "attachment; filename=Linqbooks.xml")
    context.Response.ContentType = "text/xml"
    dataSet.WriteXml(context.Response.Output)
  End Sub

  ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
    Get
      Return True
    End Get
  End Property
End Class

File: Language/C#/Book/LINQ实战/第13章 应用程序各层中的LINQ/Chapter13.CSharp - LinqBooks/LinqBooks.Web/GetXML.ashx.cs
#region Imports

using System;
using System.Data;
using System.Web;
using System.Linq;
using System.Collections;
using System.Web.Services;
using System.Web.Services.Protocols;

using LinqBooks.Web.LinqBooksDataSetTableAdapters;

#endregion Imports

namespace LinqBooks.Web
{
  [WebService(Namespace = "http://tempuri.org/")]
  [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
  public class GetXml : IHttpHandler
  {
    public void ProcessRequest(HttpContext context)
    {
      LinqBooksDataSet dataSet = new LinqBooksDataSet();
      new SubjectTableAdapter().Fill(dataSet.Subject);
      new PublisherTableAdapter().Fill(dataSet.Publisher);
      new BookTableAdapter().Fill(dataSet.Book);
      new AuthorTableAdapter().Fill(dataSet.Author);
      new BookAuthorTableAdapter().Fill(dataSet.BookAuthor);
      new UserTableAdapter().Fill(dataSet.User);
      new ReviewTableAdapter().Fill(dataSet.Review);

      context.Response.AddHeader("Content-Disposition", "attachment; filename=Linqbooks.xml");
      context.Response.ContentType = "text/xml";
      dataSet.WriteXml(context.Response.Output);
    }

    public bool IsReusable
    {
      get { return true; }
    }
  }
}
